base/collection_view.go

```diff
@@ -171,7 +171,8 @@ func (c *Collection) View(ddoc, name string, params map[string]interface{}) (sgb
 
 	if gocbViewResult != nil {
 		viewResultIterator := &gocbRawIterator{
-			rawResult: gocbViewResult,
+			rawResult:                  gocbViewResult,
```

Comment:
Need to releaseViewOp if we exit on error on line 169 I think, as well as if gocbViewResult is nil.  Since we process the entire result here, might be better to skip the gocbRawIterator approach here, and just defer close after calling executeViewQuery.

For the View case, it might be cleaner to just move the `c.waitForAvailViewOp()` up to View and ViewQuery, instead of in executeViewQuery, to reduce the code distance between acquiring and releasing the op.

Reply:
Don't think this is required? If there is an error in `executeViewQuery` we already release the view op in there.